src/sentry/testutils/asserts.py

```diff
@@ -110,3 +112,12 @@ def assert_middleware_metrics(middleware_calls):
     assert end1.args[0] == EventLifecycleOutcome.SUCCESS
     assert start2.args[0] == EventLifecycleOutcome.STARTED
     assert end2.args[0] == EventLifecycleOutcome.SUCCESS
+
+
+def assert_count_of_metric(mock_record, outcome, outcome_count):
```

Comment:
for asserting failure, it might help to be more specific by checking the message that the lifecycle is terminating with like here https://github.com/getsentry/sentry/blob/6eeb1232632d51d0cadd72445bad64fede367723/src/sentry/testutils/asserts.py#L73-L80

currently when you assert a single failure metric, you're only asserting on the count and not the exact line of code, which could be done if you check which exception or message it failed with

Reply:
we can do both, currently in the tests I do `assert_failure_metric` alongside `assert_count_of_metric` for most of the tests, but yeah I can make that consistent and do both for all.